libvirt/domain.go

```diff
@@ -230,6 +235,134 @@ func newDiskForCloudInit(virConn *libvirt.Connect, volumeKey string) (libvirtxml
 	return disk, nil
 }
 
+func getFirstDiskPath(d *schema.ResourceData, domainDef *libvirtxml.Domain, virConn *libvirt.Connect) (string, error) {
+	prefix := fmt.Sprintf("disk.%d", 0)
+
+	if volumeKey, ok := d.GetOk(prefix + ".volume_id"); ok {
+		diskVolume, err := virConn.LookupStorageVolByKey(volumeKey.(string))
+		if err != nil {
+			return "", fmt.Errorf("Error lookup storage disk volume: %s", err)
+		}
+		diskVolumeFile, err := diskVolume.GetPath()
+		if err != nil {
+			return "", fmt.Errorf("Error get path of disk volume: %s", err)
+		}
+		return diskVolumeFile, nil
+	}
+	return "", fmt.Errorf("failed to find disk 0")
+}
+
+func getStderr(stderr io.ReadCloser) string {
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(stderr)
+	return buf.String()
+}
+
+func inlineupdateCoreOSIgnition(d *schema.ResourceData, domainDef *libvirtxml.Domain, virConn *libvirt.Connect) error {
+	if ignition, ok := d.GetOk("coreos_ignition"); ok {
+		//basically we are doing following scripts
+		//eval  $(guestfish --listen -a $1)
+		//guestfish --remote -- run
+		//guestfish --remote -- findfs-label
+		//guestfish --remote -- mount $2 /
+		//guestfish --remote -- mkdir-p $3
+		//guestfish --remote -- upload $4 $3/$4
+		//guestfish --remote -- umount-all
+		//guestfish --remote -- exit
+
+		// Get the root disk (first disk) volume location
+		// FIXME: currently assume first one will be the boot disk
+		rootPath, err := getFirstDiskPath(d, domainDef, virConn)
+		if err != nil {
+			return err
+		}
+
+		// Get the ingition file to be uploaded
+		ignitionKey, err := getIgnitionVolumeKeyFromTerraformID(ignition.(string))
+		if err != nil {
+			return err
+		}
+		diskVolume, err := virConn.LookupStorageVolByKey(ignitionKey)
+		if err != nil {
+			return err
+		}
+		fileToUpload, err := diskVolume.GetPath()
+		if err != nil {
+			return err
+		}
+
+		// This is the file in coreos image going to be copied from local
+		fileRemoteLocation := "/ignition/config.ign"
+
+		cmd := exec.Command("guestfish", "--listen", "-a", rootPath)
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			return fmt.Errorf("Get stdout failed: %v", err)
+		}
+		stderr, err := cmd.StderrPipe()
+		if err != nil {
+			return fmt.Errorf("Get stderr failed: %v", err)
+		}
+
+		err = cmd.Start()
+		if err != nil {
+			return fmt.Errorf("Start command failed: %v", err)
+		}
+
+		// if the command start successfully, the output will be following format
+		// GUESTFISH_PID=xxxx; export GUESTFISH_PID, so we need get the PID and set by using os.Setenv
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(stdout)
+		s := strings.Split(buf.String(), ";")
```

Comment:
can we substitute `s` with `GUESTFISH_PID_LABEL`   
and `s1` with `GUESTFISH_PID_VALUE` something in this direction